File: apps/api/Models/Habilidade.cs
namespace api.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Table("habilidades")]
public class Habilidade
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    public int IdNivelEnsino { get; set; }
    public string Tipo { get; set; }

    public string Descricao { get; set; }

    public string? Resumo { get; set; }

    public bool Ativo { get; set; }
    public ICollection<HabilidadesXPlanejamento> HabilidadesXPlanejamentos { get; set; }

    public ICollection<Atividade> Atividades { get; set; }

}

File: apps/web/src/types/habilidade.ts
export interface Habilidade {
  id: number
  idNivelEnsino?: number
  tipo?: string
  descricao?: string
  resumo?: string
  ativo?: boolean
}

export interface HabilidadeResponse {
  sucesso: boolean
  mensagens: string[]
  objeto: Habilidade[]
}
